crates/map/src/helpers/digger.cairo

```diff
@@ -0,0 +1,245 @@
+        };
+        // [Compute] Generate the maze from the exit to the grid
+        let mut maze = Bitmap::set(0, start);
+        Mazer::iter(width, height, next, ref grid, ref maze, ref seed);
+        // [Return] The original grid with the maze to the exit
+        let grid: u256 = grid.into() | maze.into();
+        grid.try_into().unwrap()
+    }
+
+    /// Dig a corridor from the edge to the grid with a single contact point.
+    /// # Arguments
+    /// * `width` - The width of the grid
+    /// * `height` - The height of the grid
+    /// * `start` - The starting position
+    /// * `grid` - The original grid
+    /// * `seed` - The seed to generate the maze
+    /// # Returns
+    /// * The grid with the maze to the exit
+    #[inline]
+    fn corridor(width: u8, height: u8, start: u8, grid: felt252, mut seed: felt252) -> felt252 {
+        // [Check] Position is not a corner and is on an edge
+        Asserter::assert_not_corner(width, height, start);
+        Asserter::assert_on_edge(width, height, start);
+        // [Effect] Dig the edge and compute the next position
+        let (x, y) = (start % width, start / width);
+        let next: u8 = if x == 0 {
+            Mazer::next(width, start, 1)
+        } else if x == width - 1 {
+            Mazer::next(width, start, 3)
+        } else if y == 0 {
+            Mazer::next(width, start, 0)
+        } else {
+            Mazer::next(width, start, 2)
+        };
+        // [Compute] Generate the maze from the exit to the grid
+        let mut maze = Bitmap::set(0, start);
+        let mut stop = false;
+        Self::iter(width, height, next, grid, ref stop, ref maze, ref seed);
+        // [Return] The original grid with the maze to the exit
+        let grid: u256 = grid.into() | maze.into();
+        grid.try_into().unwrap()
+    }
+
+    /// Recursive function to generate the corridor on an existing grid.
+    /// # Arguments
+    /// * `width` - The width of the corridor
+    /// * `height` - The height of the corridor
+    /// * `start` - The starting position
+    /// * `grid` - The original grid
+    /// * `stop` - The stop criteria
+    /// * `maze` - The generated corridor
+    /// * `seed` - The seed to generate the corridor
+    #[inline]
+    fn iter(
+        width: u8,
+        height: u8,
+        start: u8,
+        grid: felt252,
+        ref stop: bool,
+        ref maze: felt252,
+        ref seed: felt252
+    ) {
+        // [Check] Stop criteria, the position collides with the grid
+        if Bitmap::get(grid, start) == 1 {
+            stop = true;
+            return;
+        }
+        // [Effect] Set the position
+        maze = Bitmap::set(maze, start);
+        // [Compute] Generate shuffled neighbors
+        seed = Seeder::shuffle(seed, seed);
+        let mut directions = Mazer::compute_shuffled_directions(seed);
+        // [Assess] Direction 1
+        let direction: u8 = (directions % DIRECTION_SIZE).try_into().unwrap();
```

Comment:
`(directions % DIRECTION_SIZE).try_into().unwrap()`
This is repeated here and in other files, and the result (`u32`) is always converted back to a `u8`. We may have some options here:
1. Introducing a type for `Direction` to encapsulate this logic.
2. A function doing this conversion by clarifying why it's useful to do such conversion.

Reply:
Moved everything inside the direction type